### PR TITLE
[SPARKR-228] Merge required changes from sparkr-sql to master.

### PR DESCRIPTION
**Pull request by sun-rui**
This patch fixes https://sparkr.atlassian.net/browse/SPARKR-228, which port back changes (not related to data frame)  from sparkr-sql branch to master. This patch is based on the commit 2fc553ff8c282ad0b319f444801d220ccfbfb47b on sparkr-sql branch.

### CODE REVIEW

File: pkg/R/RDD.R [hunk 1]
```diff
@@ -10,6 +10,7 @@ setOldClass("jobj")
 #'
 #' @slot env An R environment that stores bookkeeping states of the RDD
 #' @slot jrdd Java object reference to the backing JavaRDD
+#' to an RDD
```
Comment (shivaram) [inline pkg/R/RDD.R]: nit: This line can be removed

**Comment by shivaram** [comment]
Thanks @sun-rui for the merge. LGTM but for a very minor inline comment.

@davies would be good if you also took a look.

**Comment by davies** [comment]
LGTM. Almost all the changes are kind of refactor or not used (serialize Date/Time). Having a common base between these two branch will help future develop and cherry-pick, but we are going to develop in Spark, and can not merge patch from any of them directly. What's the plan for these two branches?

**Comment by shivaram** [comment]
Yeah so this merge is just a one time thing. For the next week or so we will keep the branches in sync by cherry picking changes across them (like I did yesterday for the sparkR.stop fix).  After that we will port the latest from `sparkr-sql` to amplab-extras/spark and only merge maintenance patches in master / sparkr-sql.

I know that its all a bit confusing, but hopefully its only for short term and things should become smooth after the merge

**Comment by davies** [comment]
@shivaram I had started to merge SparkR-pkg/sparkr-sql into Spark, will be available soon (today).

**Comment by shivaram** [comment]
Ok - we still need #214 and #208 is also close to being done, but I guess we can try out the cross-repo cherry-pick on them

File: pkg/R/RDD.R (hunk 1 above)
Comment (sun-rui) [inline pkg/R/RDD.R]: fixed

**Comment by shivaram** [comment]
Merging this

**Pull request merged**